### ISSUE TITLE
Checkpoints are not being saved despite `save_steps` being set

### ISSUE THREAD

**Issue by griff4692** [axolotl-ai-cloud/axolotl#843]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

With config set to

```
save_steps: 100
eval_steps: 100
save_strategy: steps
output_dir: /nlp/mistral_weights/baseline
```

I expected model checkpoints to save every 100 steps after validation.

### Current behaviour

Yet `/nlp/mistral_weights/baseline` is empty after validation and the accelerate logs show it didn't attempt to save.

### Steps to reproduce

`accelerate launch -m axolotl.cli.train ~/bhc-sum/llama/mistral_config.yml`

### Config yaml

base_model: mistralai/Mistral-7B-v0.1
model_type: MistralForCausalLM
tokenizer_type: LlamaTokenizer
is_mistral_derived_model: true

load_in_8bit: false
load_in_4bit: false
strict: false

datasets:
  - path: /nlp/projects/summarization/bhc_data_cleanup/prompt_baseline.jsonl
    type: summarizetldr

dataset_prepared_path:
val_set_size: 0.005
output_dir: /nlp/projects/summarization/bhc_data_cleanup/mistral_weights/baseline

sequence_len: 8192
sample_packing: false
pad_to_sequence_len: true

wandb_project: mistral
wandb_entity: griffinadams
wandb_watch:
wandb_run_id: baseline
wandb_log_model:

gradient_accumulation_steps: 8
micro_batch_size: 1
num_epochs: 1
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.000005

train_on_inputs: false
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 10
eval_steps: 100
eval_table_size:
eval_table_max_new_tokens: 128
save_steps: 100
save_strategy: steps
debug:
deepspeed: /home/ga2530/axolotl/deepspeed/zero2.json
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.9

### axolotl branch-commit

main/f544ab2bed513bef269e6887d35c8aa12a852473

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.
